src/transformers/generation_tf_utils.py

```diff
@@ -870,6 +875,10 @@ def _generate_no_beam_search(
             )
 
         while cur_len < max_length:
+            # past drops the `encoder_outputs` during the loop and it may be needed (encoder-decoder models)
+            if len(past) > 1 and encoder_outputs is not None:
+                past = (encoder_outputs, past)
```

Comment:
Let's double check with @patrickvonplaten for this change.

Reply:
I've reworked this part to be less intrusive: the `past` variable is not touched, `encoder_outputs` is passed to `prepare_inputs_for_generation` a few lines below. This means that it will be ignored as part of `kwargs` in most models, and encoder-decoder models can now explicitly access it if they wish to do so (like `TFSpeech2Text`).

This change also solves the `assert` comments below, in `prepare_inputs_for_generation`.